Ej/Enter project name view

lightlii opened the pull request:
Adds final view to migration (_data is detected_ flow) 

closes https://github.com/digidem/mapeo-desktop-shell/issues/8

---

# Preview 
![image](https://user-images.githubusercontent.com/32508751/232777646-da52dcea-4219-479e-8085-53413ca2bb30.png)

![image](https://user-images.githubusercontent.com/32508751/232777737-2d661f7f-e707-4e06-87e5-0df1a172502d.png)

lightlii committed:
feat: add final migration view

lightlii committed:
chore: add link from previous migration screen to final migration screen

lightlii committed:
chore: add link from previous migration screen to final migration screen

lightlii committed:
chore: add /migration-complete route to router

achou11 committed:
implement basic home layout (#23)

ErikSin commented on src/renderer/src/views/MigrationComplete/index.tsx:
```diff
@@ -0,0 +1,148 @@
+import { DefaultLayout } from '@renderer/layouts/default'
+import { useState } from 'react'
+import { defineMessages, useIntl } from 'react-intl'
+import { Link } from 'react-router-dom'
```
Can get rid off `Link`

ErikSin commented on src/renderer/src/views/MigratingProject/index.tsx:
I can't access this file right now. Once you merge your other PR and rebase this, I can re-review

ErikSin reviewed (changes requested):
5 file, new record! Thanks for keeping it scoped :)

lightlii committed:
Ej/navigation view (#36)

* feat: add react-intl and setup basic internationalisation

* chore: add nvmrc

* refactor: extract IntlProvider and lang switcher logic into own component

* fix: import IntlSwitch properly, better messages example id

* fix: run extract-messages to update IDs in generated messages file

* fix: remove duplication in gitignore

* chore: run extract-messages pre-commit

* chore: reverse typeing translations

* feat: add reach router + basic router component

* chore: remove template 'Versions' component

* chore: cleaning

* chore: add mui

* feat: add lang switcher, default layout

* feat: add Language switcher

* feat: add theme.ts for keeping colours etc centralised

* chore: add icons-material package

* Ej/Scaffold app + internationalisation (#4)

* chore: add extract-messages script

* fix: fix app title in html file 😅

* feat: add react-intl and setup basic internationalisation

* chore: add nvmrc

* refactor: extract IntlProvider and lang switcher logic into own component

* fix: import IntlSwitch properly, better messages example id

* fix: run extract-messages to update IDs in generated messages file

* fix: remove duplication in gitignore

* chore: run extract-messages pre-commit

* chore: reverse typeing translations

* chore: remove yarn.lock

* chore: gitignore yarn.lock, generate package-lock.json

* chore: add compile script to generate translations files

* chore: add quotes on compile script

* chore: update compile script / move translations

* chore: update IntlProvider component to filter from available languages

* chore: set default lang to 'es'

* feat: add react-intl and setup basic internationalisation

* refactor: extract IntlProvider and lang switcher logic into own component

* feat: add lang switcher, default layout

* feat: add language select component

* feat: add simple splash screen

* feat: add fonts + fontface rules

* feat: add theme and theme provider

* feat: add initial migration view

* chore: add variables as constants

* chore: further styling to init screen

* feat: add skip-migration modal

* chore: add auto-generated translations

* fix: replace reach-router with react-router

* feat: add navigation screen

* feat: more work on 'navigation' view

* chore: move Logo component to components directory, add Mapeo5 Logo

* feat: add Migration view

* chore: cleanup

* chore: remove mapeo5 logo once migration complete

* chore: set modal size to fill most of screen

* chore: new modal complete state layout

* chore: add views to 'extract-messages' script

* Squashed commit of the following:

commit 132b8164ba8005d4b97897e42206ad546c0662b4
Author: lightlii <32508751+lightlii@users.noreply.github.com>
Date:   Wed Apr 12 12:09:48 2023 +0200

    Ej/initial screen layout (#10)

    * feat: add react-intl and setup basic internationalisation

    * refactor: extract IntlProvider and lang switcher logic into own component

    * chore: cleaning

    * Ej/Scaffold app + internationalisation (#4)

    * chore: add extract-messages script

    * fix: fix app title in html file 😅

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * chore: remove yarn.lock

    * chore: gitignore yarn.lock, generate package-lock.json

    * chore: add compile script to generate translations files

    * chore: add quotes on compile script

    * chore: update compile script / move translations

    * chore: update IntlProvider component to filter from available languages

    * chore: set default lang to 'es'

    * feat: add lang switcher, default layout

    * feat: add language select component

    * feat: add simple splash screen

    * feat: add fonts + fontface rules

    * feat: add theme and theme provider

    * feat: add initial migration view

    * chore: add variables as constants

    * chore: further styling to init screen

    * chore: remove fonts

    * Ej/Scaffold app + internationalisation (#4)

    * chore: add extract-messages script

    * fix: fix app title in html file 😅

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * chore: remove yarn.lock

    * chore: gitignore yarn.lock, generate package-lock.json

    * chore: add compile script to generate translations files

    * chore: add quotes on compile script

    * chore: update compile script / move translations

    * chore: update IntlProvider component to filter from available languages

    * chore: set default lang to 'es'

    * Ej/More setup (#9)

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * feat: add reach router + basic router component

    * chore: remove template 'Versions' component

    * chore: cleaning

    * chore: add mui

    * feat: add lang switcher, default layout

    * feat: add Language switcher

    * feat: add theme.ts for keeping colours etc centralised

    * chore: add icons-material package

    * Ej/Scaffold app + internationalisation (#4)

    * chore: add extract-messages script

    * fix: fix app title in html file 😅

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * chore: remove yarn.lock

    * chore: gitignore yarn.lock, generate package-lock.json

    * chore: add compile script to generate translations files

    * chore: add quotes on compile script

    * chore: update compile script / move translations

    * chore: update IntlProvider component to filter from available languages

    * chore: set default lang to 'es'

    * feat: add react-intl and setup basic internationalisation

    * refactor: extract IntlProvider and lang switcher logic into own component

    * feat: add lang switcher, default layout

    * feat: add language select component

    * feat: add simple splash screen

    * chore: remove unused imports

    * chore: remove redundant react fragment

    * chore: set global CSS colours from theme variables

    * chore: clean up intl provier tertiary

    * chore: remove timer hook

    ---------

    Co-authored-by: ErikSin <67773827+ErikSin@users.noreply.github.com>

    * feat: add react-intl and setup basic internationalisation

    * Ej/Scaffold app + internationalisation (#4)

    * chore: add extract-messages script

    * fix: fix app title in html file 😅

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * chore: remove yarn.lock

    * chore: gitignore yarn.lock, generate package-lock.json

    * chore: add compile script to generate translations files

    * chore: add quotes on compile script

    * chore: update compile script / move translations

    * chore: update IntlProvider component to filter from available languages

    * chore: set default lang to 'es'

    * refactor: extract IntlProvider and lang switcher logic into own component

    * feat: add lang switcher, default layout

    * feat: add fonts + fontface rules

    * feat: add theme and theme provider

    * feat: add initial migration view

    * chore: rebase fixes

    * fix: rename themeVarient => langBackgroundVarient for clarity

    * chore: remove all fonts again (?? rebase reinstated them 😑)

    * chore: remove button component

    * fix: remove hardcoded css again

    * chore: add fonts (#19)

    * fix: import useTheme (missing from rebase)

    * fix: remove fonts again (help)

    ---------

    Co-authored-by: ErikSin <67773827+ErikSin@users.noreply.github.com>

commit 0a546ca7c0fc87e7d5928ecf20cab74f419b43e8
Author: lightlii <32508751+lightlii@users.noreply.github.com>
Date:   Wed Apr 5 11:39:18 2023 +0200

    Ej/More setup (#9)

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * feat: add reach router + basic router component

    * chore: remove template 'Versions' component

    * chore: cleaning

    * chore: add mui

    * feat: add lang switcher, default layout

    * feat: add Language switcher

    * feat: add theme.ts for keeping colours etc centralised

    * chore: add icons-material package

    * Ej/Scaffold app + internationalisation (#4)

    * chore: add extract-messages script

    * fix: fix app title in html file 😅

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * chore: remove yarn.lock

    * chore: gitignore yarn.lock, generate package-lock.json

    * chore: add compile script to generate translations files

    * chore: add quotes on compile script

    * chore: update compile script / move translations

    * chore: update IntlProvider component to filter from available languages

    * chore: set default lang to 'es'

    * feat: add react-intl and setup basic internationalisation

    * refactor: extract IntlProvider and lang switcher logic into own component

    * feat: add lang switcher, default layout

    * feat: add language select component

    * feat: add simple splash screen

    * chore: remove unused imports

    * chore: remove redundant react fragment

    * chore: set global CSS colours from theme variables

    * chore: clean up intl provier tertiary

    * chore: remove timer hook

    ---------

    Co-authored-by: ErikSin <67773827+ErikSin@users.noreply.github.com>

commit 1460c04e186edeab5f24c7a84e98b262bc22895c
Merge: aa5b370 2c13648
Author: lightlii <32508751+lightlii@users.noreply.github.com>
Date:   Thu Mar 30 17:50:32 2023 +0200

    Merge branch 'develop' of github.com:digidem/mapeo-desktop-shell into develop

commit 2c13648809961197c436c55991761405ce20b346
Author: lightlii <32508751+lightlii@users.noreply.github.com>
Date:   Thu Mar 30 17:40:05 2023 +0200

    Ej/Scaffold app + internationalisation (#4)

    * chore: add extract-messages script

    * fix: fix app title in html file 😅

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * chore: remove yarn.lock

    * chore: gitignore yarn.lock, generate package-lock.json

    * chore: add compile script to generate translations files

    * chore: add quotes on compile script

    * chore: update compile script / move translations

    * chore: update IntlProvider component to filter from available languages

    * chore: set default lang to 'es'

* fix: remove fonts

* chore: remove redundant useEffect and clear state on modal unmoutn

* chore: simplified button component (removed loading + disabled states)

* chore: remove redundant useState for progress variable

* chore: remove redundant useRef for `oberservations` and `media` variables

* chore: remove fonts

* fix: reinstate missing MigratingProject file (??), fix default layout

* chore: add auto-generated translation file

* fix: fix migratingProject view layout

* feat: add index navigation view

* chore: bold h1 title

* feat: add menu icon button to default layout to return to navigation

* chore: remove nav menu icon for now

* Ej/ fix migrating layout (#35)

* feat: add react-intl and setup basic internationalisation

* chore: add nvmrc

* refactor: extract IntlProvider and lang switcher logic into own component

* fix: import IntlSwitch properly, better messages example id

* fix: run extract-messages to update IDs in generated messages file

* fix: remove duplication in gitignore

* chore: run extract-messages pre-commit

* chore: reverse typeing translations

* feat: add reach router + basic router component

* chore: remove template 'Versions' component

* chore: cleaning

* chore: add mui

* feat: add lang switcher, default layout

* feat: add Language switcher

* feat: add theme.ts for keeping colours etc centralised

* chore: add icons-material package

* Ej/Scaffold app + internationalisation (#4)

* chore: add extract-messages script

* fix: fix app title in html file 😅

* feat: add react-intl and setup basic internationalisation

* chore: add nvmrc

* refactor: extract IntlProvider and lang switcher logic into own component

* fix: import IntlSwitch properly, better messages example id

* fix: run extract-messages to update IDs in generated messages file

* fix: remove duplication in gitignore

* chore: run extract-messages pre-commit

* chore: reverse typeing translations

* chore: remove yarn.lock

* chore: gitignore yarn.lock, generate package-lock.json

* chore: add compile script to generate translations files

* chore: add quotes on compile script

* chore: update compile script / move translations

* chore: update IntlProvider component to filter from available languages

* chore: set default lang to 'es'

* feat: add react-intl and setup basic internationalisation

* refactor: extract IntlProvider and lang switcher logic into own component

* feat: add lang switcher, default layout

* feat: add language select component

* feat: add simple splash screen

* feat: add fonts + fontface rules

* feat: add theme and theme provider

* feat: add initial migration view

* chore: add variables as constants

* chore: further styling to init screen

* feat: add skip-migration modal

* chore: add auto-generated translations

* fix: replace reach-router with react-router

* feat: add navigation screen

* feat: more work on 'navigation' view

* chore: move Logo component to components directory, add Mapeo5 Logo

* feat: add Migration view

* chore: cleanup

* chore: remove mapeo5 logo once migration complete

* chore: set modal size to fill most of screen

* chore: new modal complete state layout

* chore: add views to 'extract-messages' script

* Squashed commit of the following:

commit 132b8164ba8005d4b97897e42206ad546c0662b4
Author: lightlii <32508751+lightlii@users.noreply.github.com>
Date:   Wed Apr 12 12:09:48 2023 +0200

    Ej/initial screen layout (#10)

    * feat: add react-intl and setup basic internationalisation

    * refactor: extract IntlProvider and lang switcher logic into own component

    * chore: cleaning

    * Ej/Scaffold app + internationalisation (#4)

    * chore: add extract-messages script

    * fix: fix app title in html file 😅

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * chore: remove yarn.lock

    * chore: gitignore yarn.lock, generate package-lock.json

    * chore: add compile script to generate translations files

    * chore: add quotes on compile script

    * chore: update compile script / move translations

    * chore: update IntlProvider component to filter from available languages

    * chore: set default lang to 'es'

    * feat: add lang switcher, default layout

    * feat: add language select component

    * feat: add simple splash screen

    * feat: add fonts + fontface rules

    * feat: add theme and theme provider

    * feat: add initial migration view

    * chore: add variables as constants

    * chore: further styling to init screen

    * chore: remove fonts

    * Ej/Scaffold app + internationalisation (#4)

    * chore: add extract-messages script

    * fix: fix app title in html file 😅

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * chore: remove yarn.lock

    * chore: gitignore yarn.lock, generate package-lock.json

    * chore: add compile script to generate translations files

    * chore: add quotes on compile script

    * chore: update compile script / move translations

    * chore: update IntlProvider component to filter from available languages

    * chore: set default lang to 'es'

    * Ej/More setup (#9)

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * feat: add reach router + basic router component

    * chore: remove template 'Versions' component

    * chore: cleaning

    * chore: add mui

    * feat: add lang switcher, default layout

    * feat: add Language switcher

    * feat: add theme.ts for keeping colours etc centralised

    * chore: add icons-material package

    * Ej/Scaffold app + internationalisation (#4)

    * chore: add extract-messages script

    * fix: fix app title in html file 😅

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * chore: remove yarn.lock

    * chore: gitignore yarn.lock, generate package-lock.json

    * chore: add compile script to generate translations files

    * chore: add quotes on compile script

    * chore: update compile script / move translations

    * chore: update IntlProvider component to filter from available languages

    * chore: set default lang to 'es'

    * feat: add react-intl and setup basic internationalisation

    * refactor: extract IntlProvider and lang switcher logic into own component

    * feat: add lang switcher, default layout

    * feat: add language select component

    * feat: add simple splash screen

    * chore: remove unused imports

    * chore: remove redundant react fragment

    * chore: set global CSS colours from theme variables

    * chore: clean up intl provier tertiary

    * chore: remove timer hook

    ---------

    Co-authored-by: ErikSin <67773827+ErikSin@users.noreply.github.com>

    * feat: add react-intl and setup basic internationalisation

    * Ej/Scaffold app + internationalisation (#4)

    * chore: add extract-messages script

    * fix: fix app title in html file 😅

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * chore: remove yarn.lock

    * chore: gitignore yarn.lock, generate package-lock.json

    * chore: add compile script to generate translations files

    * chore: add quotes on compile script

    * chore: update compile script / move translations

    * chore: update IntlProvider component to filter from available languages

    * chore: set default lang to 'es'

    * refactor: extract IntlProvider and lang switcher logic into own component

    * feat: add lang switcher, default layout

    * feat: add fonts + fontface rules

    * feat: add theme and theme provider

    * feat: add initial migration view

    * chore: rebase fixes

    * fix: rename themeVarient => langBackgroundVarient for clarity

    * chore: remove all fonts again (?? rebase reinstated them 😑)

    * chore: remove button component

    * fix: remove hardcoded css again

    * chore: add fonts (#19)

    * fix: import useTheme (missing from rebase)

    * fix: remove fonts again (help)

    ---------

    Co-authored-by: ErikSin <67773827+ErikSin@users.noreply.github.com>

commit 0a546ca7c0fc87e7d5928ecf20cab74f419b43e8
Author: lightlii <32508751+lightlii@users.noreply.github.com>
Date:   Wed Apr 5 11:39:18 2023 +0200

    Ej/More setup (#9)

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * feat: add reach router + basic router component

    * chore: remove template 'Versions' component

    * chore: cleaning

    * chore: add mui

    * feat: add lang switcher, default layout

    * feat: add Language switcher

    * feat: add theme.ts for keeping colours etc centralised

    * chore: add icons-material package

    * Ej/Scaffold app + internationalisation (#4)

    * chore: add extract-messages script

    * fix: fix app title in html file 😅

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * chore: remove yarn.lock

    * chore: gitignore yarn.lock, generate package-lock.json

    * chore: add compile script to generate translations files

    * chore: add quotes on compile script

    * chore: update compile script / move translations

    * chore: update IntlProvider component to filter from available languages

    * chore: set default lang to 'es'

    * feat: add react-intl and setup basic internationalisation

    * refactor: extract IntlProvider and lang switcher logic into own component

    * feat: add lang switcher, default layout

    * feat: add language select component

    * feat: add simple splash screen

    * chore: remove unused imports

    * chore: remove redundant react fragment

    * chore: set global CSS colours from theme variables

    * chore: clean up intl provier tertiary

    * chore: remove timer hook

    ---------

    Co-authored-by: ErikSin <67773827+ErikSin@users.noreply.github.com>

commit 1460c04e186edeab5f24c7a84e98b262bc22895c
Merge: aa5b370 2c13648
Author: lightlii <32508751+lightlii@users.noreply.github.com>
Date:   Thu Mar 30 17:50:32 2023 +0200

    Merge branch 'develop' of github.com:digidem/mapeo-desktop-shell into develop

commit 2c13648809961197c436c55991761405ce20b346
Author: lightlii <32508751+lightlii@users.noreply.github.com>
Date:   Thu Mar 30 17:40:05 2023 +0200

    Ej/Scaffold app + internationalisation (#4)

    * chore: add extract-messages script

    * fix: fix app title in html file 😅

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * chore: remove yarn.lock

    * chore: gitignore yarn.lock, generate package-lock.json

    * chore: add compile script to generate translations files

    * chore: add quotes on compile script

    * chore: update compile script / move translations

    * chore: update IntlProvider component to filter from available languages

    * chore: set default lang to 'es'

* fix: remove fonts

* chore: remove redundant useEffect and clear state on modal unmoutn

* chore: simplified button component (removed loading + disabled states)

* chore: remove redundant useState for progress variable

* chore: remove redundant useRef for `oberservations` and `media` variables

* chore: remove fonts

* fix: reinstate missing MigratingProject file (??), fix default layout

* chore: add auto-generated translation file

* Ej/fix  migrating layout2 (#37)

* fix: changes to interval useEffect => ref

(merged before I'd pushed changes)

* implement basic home layout (#23)

* fix: double import from merge

---------

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>

lightlii committed:
fix: remove unused import (Link)

lightlii committed:
fix: remove duplicate keys in translation file (merge bug)

lightlii committed:
fix: autogenerated translations (reordering)

lightlii committed:
Merge branch 'develop' into ej/migration-data-detected

ErikSin commented:
Can you take a look at this again? There are some weird formatting thing happening, probably from the merge/rebase.

<img width="1071" alt="image" src="https://user-images.githubusercontent.com/67773827/234084433-9d7b591e-4233-4460-abc7-0ba83a3f2f33.png">

lightlii committed:
Merge branch 'develop' into ej/migration-data-detected

lightlii committed:
feat: add navigate to Home screen in Observations mode

lightlii committed:
fix bg colour on migration screen

lightlii commented:
that should be fixed now @ErikSin

ErikSin committed:
simplified types and reset state
